src/wechaty_puppet_hostie/puppet.py

```diff
@@ -537,7 +537,7 @@ def self_id(self) -> str:
             contact_id
         :return:
         """
-        return ''
+        return self.login_user_id
```

Comment:
`login_user_id` is Optional str, we should check the null status before return it. 
```python
if not self.login_user_id:
    raise ValueError('can"t call self_id() before logined')
return self.login_user_id
```

Reply:
Get it. login_user_id is initialized as None and the function needs type of str as return.